app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -81,13 +82,34 @@ class ThreadAdapter : RecyclerView.Adapter<ThreadViewHolder>(), RealmChangesBind
     override fun onBindViewHolder(holder: ThreadViewHolder, position: Int): Unit = with(holder.binding) {
         val message = messages[position]
 
+        setStyleIfSingleMail(position)
+
         holder.bindHeader(message)
         holder.bindAttachment(message)
         loadBodyInWebView(message.body)
 
         displayExpandedCollapsedMessage(message)
     }
 
+    private fun ItemMessageBinding.setStyleIfSingleMail(position: Int) {
+        if (itemCount == 1) {
+            root.apply {
+                setMarginsRelative(0, 0, 0, 0)
+                radius = 0f
+            }
+        } else {
+            root.apply {
+                val vertical = resources.getDimension(RCore.dimen.marginStandardVerySmall).toInt()
+                val horizontal = resources.getDimension(RCore.dimen.marginStandardSmall).toInt()
+                val topMargin = if (position == 0) 2 * vertical else vertical
+                val bottomMargin = if (position == lastIndex()) 2 * vertical else vertical
+                setMarginsRelative(horizontal, topMargin, horizontal, bottomMargin)
+
+                radius = 8.toPx().toFloat()
+            }
+        }
+    }
```

Comment:
Change it to be an extension of `MaterialCardView`, this will avoid doing `root.apply`

Modify `bindHeader` and `bindAttachment ` accordingly

Reply:
I made `setStyleIfSingleMail` into an extensions of `MaterialCardView` but I don't know what modifications you want for `bindHeader` and `bindAttachment`

Reply:
Directly an extension of `ItemMessageBinding` like other